.github/workflows/_publish_badge.yaml

```diff
@@ -0,0 +1,77 @@
+name: ~publish test results into a json file that can be used to make badge
+
+on:
+  workflow_call:
+    inputs:
+      SCRIPT:
+        type: string
+        description: 'Script to run :'
+        required: true
+      BADGE_LABEL:
+        type: string
+        description: 'Badge label:'
+        required: true
+      BADGE_FILENAME:
+        type: string
+        description: 'Filename to find the json in the gist for creating a badge:'
+        required: true
+      ARTIFACTS:
+        type: string
+        description: 'Paths of artifacts assuming they are all downloaded (globs can be used, but it is up to script to support):'
+        required: true
+        default: '**/*.jsnl'
+      PUBLISH:
+        type: boolean
+        description: Update status badge?
+        default: false
+        required: true
+
+env:
+  LOCAL_TEST_LOG_DIR: .test-logs
```

Comment:
Since it is a standalone (albeit reusable) workflow, it always starts in a clean docker container so we can simply use the current working folder to store all the downloaded artifacts. There is no risk of interference or clutter with previous jobs.